### PR TITLE
fix: Fixed edittext issue

### LINKED ISSUE #65: Issue when edit text is empty.
**Describe the bug**
When the edit text is empty, the keyboard goes down and no alert is shown.

**To Reproduce**
Keep the edit text empty and press send. The keyboard goes down.

**Expected behavior**
The keyboard should not go down and an alert should be shown in the edit text.

**Smartphone Info:**

|               | |
|---------------|-|
|Device         |Nokia 7 plus|
|Android Version|Android 9,0  |

**Additional context**
A hint shall also be added to the edit text. 

Update: Now the preview button has been introduced. Changes have to made for it also.

**Would you like to work on the issue?**
YES

### PR DESCRIPTION
**Pull request by yashk2000**
Fixes #65

Changes: The edit text now shows an error message when it is empty and send is pressed. Also, the keyboard continues to be visible. The edit text also shows a hint now when it si empty.

Screenshots for the change:

screencap

When send is pressed and edit text is empty:

screencap1

### CODE REVIEW

**Comment by liveHarshit** [comment]
Can you please check why Travis is pending for all PR?

**Comment by yashk2000** [comment]
> Move hide soft keyboard logic to utils class. If utils class is not there create a new one as an object.

I will do that and update the pull request.

**Comment by yashk2000** [comment]
> Can you please check why Travis is pending for all PR?

I do not know how to check this. When I open the Travis website, it is just showing "job received".

**Comment by yashk2000** [comment]
@liveHarshit please review. I have made the changes.

File: app/src/main/java/com/nilhcem/blenamebadge/ui/message/MessageActivity.kt [hunk 1]
```diff
@@ -45,41 +47,45 @@ class MessageActivity : AppCompatActivity() {
...
     private val mode: Spinner by bindView(R.id.mode)
     private val send: Button by bindView(R.id.send_button)
-
+    private lateinit var rootView: View
     private val presenter by lazy { MessagePresenter() }
 
     override fun onCreate(savedInstanceState: Bundle?) {
         super.onCreate(savedInstanceState)
         setContentView(R.layout.message_activity)
+        rootView = window.decorView.rootView
```
Comment (liveHarshit) [inline app/src/main/java/com/nilhcem/blenamebadge/ui/message/MessageActivity.kt]: No need to initialize rootView. Just pass window.decorView for View.

File: app/src/main/java/com/nilhcem/blenamebadge/core/utils/Utils.kt [hunk 2]
```diff
@@ -0,0 +1,15 @@
...
+import android.view.inputmethod.InputMethodManager
+
+object Utils {
+
+    fun hideSoftKeyboard(context: Context?, view: View) {
+        val inputManager: InputMethodManager = context?.getSystemService(Context.INPUT_METHOD_SERVICE)
+                as InputMethodManager
+        inputManager.hideSoftInputFromWindow(view.windowToken, InputMethodManager.SHOW_FORCED)
+    }
+
```
Comment (liveHarshit) [inline app/src/main/java/com/nilhcem/blenamebadge/core/utils/Utils.kt]: extra

File: app/src/main/res/values/strings.xml [hunk 3]
```diff
@@ -22,4 +22,6 @@
     <string name="send_button">SEND</string>
     <string name="permission_required">Permission required</string>
     <string name="txt_turn_on_bluetooth">Turn on Bluetooth and try again</string>
+    <string name="empty_message">Please enter the message to be displayed</string>
```
Comment (liveHarshit) [inline app/src/main/res/values/strings.xml]: enter a message*

File: app/src/main/java/com/nilhcem/blenamebadge/ui/message/MessageActivity.kt [hunk 4]
```diff
@@ -57,29 +57,32 @@ class MessageActivity : AppCompatActivity() {
...
-                send.isClickable = false
-                val buttonTimer = Timer()
-                buttonTimer.schedule(object : TimerTask() {
-                    override fun run() {
-                        runOnUiThread { send.isClickable = true }
+            if (content.text.isEmpty()) {
+                content.isFocusable = true
+                content.error = getString(R.string.empty_message)
+            } else {
+                Utils.hideSoftKeyboard(baseContext, window.decorView.rootView)
```
Comment (liveHarshit) [inline app/src/main/java/com/nilhcem/blenamebadge/ui/message/MessageActivity.kt]: What is rootView here?

Comment (liveHarshit) [reply app/src/main/java/com/nilhcem/blenamebadge/ui/message/MessageActivity.kt]: Can't you directly pass window.decorView?

**Comment by yashk2000** [comment]
Is it alright now @liveHarshit ?

File: app/src/main/res/values/strings.xml (hunk 3 above)
Comment (liveHarshit) [reply app/src/main/res/values/strings.xml]: Hey, I'm talking about empty_message string. But you change the another one. Sorry for the confusion.

File: app/src/main/java/com/nilhcem/blenamebadge/ui/message/MessageActivity.kt (hunk 4 above)
Comment (liveHarshit) [reply app/src/main/java/com/nilhcem/blenamebadge/ui/message/MessageActivity.kt]: But it's okay.

**Comment by liveHarshit** [comment]
Travis was failing earlier. Let's see.

**Comment by yashk2000** [comment]
> Travis was failing earlier. Let's see.

That was because of an unused import.

**Pull request closed without merging**